src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -193,6 +200,117 @@ private List<string> ResolveByExternalReferencePackages(List<string> uidList, Co
+            for (int i = 0; i < uidList.Count; i += pieceSize)
+            {
+                List<string> smallPiece;
+                smallPiece = uidList.GetRange(i, Math.Min(pieceSize, uidList.Count - i));
+                bool isSuccess = false;
+
+                foreach(string requestUrl in XRefServiceUrls)
+                {
+                    var result = await QueryByHttpRequestAsync(requestUrl, smallPiece, xsList);
+                    bool isExist = result.Item1;
+                    if(isExist)
+                    {
+                        isSuccess = true;
+                        xsList = result.Item2;
+                        break;
+                    }
+                }
+                
+                if(isSuccess)
+                {
+                    for(int j = 0; j < xsList.Count; j++)
+                    {
+                        if(xsList[j] != null)
+                        {
+                            externalXRefSpec.AddOrUpdate(smallPiece[j], xsList[j], (_, old) => old + xsList[j]);
+                        }
+                        else
+                        {
+                            unResolvedUidList.Add(smallPiece[j]);
+                        }
+                    }
+                }
+                else
+                {
+                    unResolvedUidList.AddRange(smallPiece);
+                    Logger.LogWarning($"Failed to resolve {smallPiece.Count} uids from all requestUrls: for example include " + smallPiece.Take(10).ToDelimitedString());
+                }
+            }
+            
+            Logger.LogInfo($"{uidList.Count - unResolvedUidList.Count} external references found in xrefservice configured in docfx.json");
+            return unResolvedUidList;
+        }
+
+        private async Task<Tuple<bool, List<XRefSpec>>> QueryByHttpRequestAsync(string requestUrl, List<string> smallPiece, List<XRefSpec> xsList)
+        {
+            using (var client = new HttpClient())
+            {
+                var failedResult = new Tuple<bool, List<XRefSpec>>(false, null);
+                try
+                {
+                    client.BaseAddress = new Uri(requestUrl);
+                }
+                catch (UriFormatException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return failedResult;
+                }
+                catch (ArgumentException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return failedResult;
+                }
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+                StringContent content = new StringContent(JsonUtility.Serialize(smallPiece), System.Text.Encoding.UTF8, "application/json");
```

Comment:
i don't quite understand? can you give an example?

---
In reply to: [127136214](https://github.com/dotnet/docfx/pull/1837#discussion_r127136214) [](ancestors = 127136214)